src/mlpack/core/dists/diagonal_gaussian_distribution.hpp

```diff
@@ -18,15 +18,16 @@ namespace mlpack {
 namespace distribution {
 
 //! A single multivariate Gaussian distribution with diagonal covariance.
-class DiagonalGaussianDistribution
+template<typename MatType, typename Vectype>
```

Comment:
Is there a situation where `Vectype !=  arma::Col<typename DataType::elem_type>`? I guess if there is not we could remove the second template parameter? Also would be nice to document the template parameter.